crates/rust-analyzer/src/global_state.rs

```diff
@@ -126,6 +126,10 @@ pub(crate) struct GlobalState {
         OpQueue<(), (Arc<Vec<ProjectWorkspace>>, Vec<anyhow::Result<WorkspaceBuildScripts>>)>,
     pub(crate) fetch_proc_macros_queue: OpQueue<Vec<ProcMacroPaths>, bool>,
     pub(crate) prime_caches_queue: OpQueue,
+
+    /// a deferred task queue. this should only be used if the enqueued Task
+    /// can only run *after* [`GlobalState::process_changes`] has been called.
```

Comment:
A deferred task queue used for doing database dependent work inside sync handlers. Accessing the database may block and as such should be offloaded from the main thread.

Reply:
i’ll incorporate that. at least for this functionality, the fact that this occurs _after_ `GlobalState::process_changes` is load-bearing, so i’ll keep that detail.

(it’s also enforced by the slow test I added, but it doesn’t hurt to restate this detail.)

Reply:
done!